src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -0,0 +1,217 @@
+    <_GenerateResourceDesignerAssemblyOutput>$(_DesignerIntermediateOutputPath)$(_DesignerAssemblyName).dll</_GenerateResourceDesignerAssemblyOutput>
+    <_GenerateResourceDesignerClassFile Condition=" '$(Language)' == 'F#' ">$(_DesignerIntermediateOutputPath)_$(_DesignerAssemblyName).fs</_GenerateResourceDesignerClassFile>
+    <_GenerateResourceDesignerClassFile Condition=" '$(_GenerateResourceDesignerClassFile)' == '' ">$(_DesignerIntermediateOutputPath)_$(_DesignerAssemblyName).cs</_GenerateResourceDesignerClassFile>
+    <_GenerateResourceCaseMapFile>$(_DesignerIntermediateOutputPath)case_map.txt</_GenerateResourceCaseMapFile>
+  </PropertyGroup>
+  <Message Text="_OuterIntermediateOutputPath: $(_OuterIntermediateOutputPath)" />
+  <Message Text="IntermediateOutputPath: $(IntermediateOutputPath)" />
+</Target>
+
+<Target Name="_GenerateRtxt"
+    Condition="'$(AndroidUseDesignerAssembly)' == 'True' And '$(DesignTimeBuild)' == 'True' "
+    DependsOnTargets="_CreatePropertiesCache;_ResolveSdks;_ResolveAndroidTooling;_GetJavaPlatformJar;_GenerateAndroidResourceDir;_SetupDesignerProperties"
+    Inputs="$(_AndroidResFlagFile);@(_AndroidResourceDest);@(LibraryResourceDirectories->'%(StampFile)')"
+    Outputs="$(_DesignerIntermediateOutputPath)R.txt"
+  >
+    <!-- Generate an R.txt file using the Managed Parser -->
+    <GenerateRtxt
+        AdditionalResourceDirectories="@(LibraryResourceDirectories)"
+        CaseMapFile="$(_GenerateResourceCaseMapFile)"
+        JavaPlatformJarPath="$(JavaPlatformJarPath)"
+        ResourceDirectory="$(MonoAndroidResDirIntermediate)"
+        ResourceFlagFile="$(_AndroidResFlagFile)"
+        RTxtFile="$(_DesignerIntermediateOutputPath)R.txt"
+    >
+    </GenerateRtxt>
+</Target>
+
+<Target Name="_GenerateResourceCaseMap"
+    DependsOnTargets="_ComputeAndroidResourcePaths;_SetupDesignerProperties;_GetLibraryImports"
+    Inputs="@(_AndroidResourceDest);@(LibraryResourceDirectories->'%(StampFile)')"
+    Outputs="$(_GenerateResourceCaseMapFile)"
+  >
+    <!-- Generate a ResourceMap file for the project and its resources -->
+    <GenerateResourceCaseMap
+        AdditionalResourceDirectories="@(LibraryResourceDirectories)"
+        OutputFile="$(_GenerateResourceCaseMapFile)"
+        ProjectDir="$(ProjectDir)"
+        ResourceDirectory="$(MonoAndroidResDirIntermediate)"
+        Resources="@(_AndroidResourceDest)"
+    />
+    <Touch Files="$(_GenerateResourceCaseMapFile)" />
+    <ItemGroup>
+      <FileWrites Include="$(_GenerateResourceCaseMapFile)" />
+    </ItemGroup>
+</Target>
+
+<!--
+    We only want to do this for non application projects unless the user has
+    specifically opted into the new system. This can be done by either
+    1. using `dotnet new android` and building a new project.
+    2. Add <AndroidUseDesignerAssembly>true</AndroidUseDesignerAssembly> to
+       the project.
```

Comment:
Does this comment match the `.md` file above? (Which says `AndroidUseDesignerAssembly=true` by default) Should it say one thing for classic Xamarin.Android, and one thing for .NET 8?